app/models/collection.rb

```diff
@@ -146,8 +146,25 @@ def work_type
     'collection'
   end
 
+  def thumbnail_url
+    auto_generate_thumbnail? ? auto_generated_thumbnail_url : nil
+  end
+
+  def auto_generated_thumbnail_url
+    thumbnail_urls.last
+  end
+
+  def thumbnail_present?
+    thumbnail_urls.present?
+  end
```

Comment:
I could go either way on this.  On one hand, we have duplication that we could resolve with this abstraction.  On the other hand, some of these features are still developing, and the abstraction might be a bit premature.  This is my inexperienced opinion, though.  @rschenk would you create a module here?

Reply:
I agree with both of you. To Nick's point, yes, a Thumbnailable module would be the Ruby way to share these implementations. To Alex's point, Adam and I often had a rule of thumb to not go too nuts on refactoring until the third time we were doing something. Alex I like how you used an rspec shared example group to at least enforce the contract between all the thumbnailable things, which I think is good enough for now. Maybe it would warrant a little comment on the various thumbnailable classes pointing out that it's a good place for a future refactor?